src/vs/workbench/api/browser/mainThreadDebugService.ts

```diff
@@ -337,7 +337,7 @@ export class MainThreadDebugService implements MainThreadDebugServiceShape, IDeb
 		return undefined;
 	}
 
-	private convertToDto(bps: (ReadonlyArray<IBreakpoint | IFunctionBreakpoint | IDataBreakpoint>)): Array<ISourceBreakpointDto | IFunctionBreakpointDto | IDataBreakpointDto> {
+	private convertToDto(bps: (ReadonlyArray<IBreakpoint | IFunctionBreakpoint | IDataBreakpoint | IInstructionBreakpoint>)): Array<ISourceBreakpointDto | IFunctionBreakpointDto | IDataBreakpointDto> {
```

Comment:
This should also have the `IInstructionBreakpoint` as the return type.
Also make sure the conversion is properly done please.

Reply:
@isidorn supporting instruction breakpoints in the debug extension API requires more work (which is not part of this PR)